### PR TITLE
Managing custom headers for Swiftmailer

### PR DESCRIPTION
**Pull request by TristanPouliquen**
This PR adds the saving of custom [unstructured headers](https://github.com/swiftmailer/swiftmailer/blob/85b7e7d12e2d8d897bdba1e0a6496e2c6d700164/lib/classes/Swift/Mime/Headers/UnstructuredHeader.php) to the spool database and the outputting of said headers to the Swift_Message instance sent to [WhiteOctoberSwiftMailerDBBundle DatabaseSpool](https://github.com/whiteoctober/WhiteOctoberSwiftMailerDBBundle/blob/master/Spool/DatabaseSpool.php).

Restricted to unstructured headers to limit code complexity (could be extended in the future if the needs arise) and because text headers (common custom headers) are managed as UnstructuredHeaders.

### CODE REVIEW

**Comment by TristanPouliquen** [comment]
Pull request following issue #13

File: composer.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-    "name": "appventus/spoolmailerbundle",
+    "name": "TristanPouliquen/spoolmailerbundle",
```
Comment (lenybernard) [inline composer.json]: can you please revert this modification ?

File: Entity/Mail.php [hunk 2]
```diff
@@ -197,7 +206,15 @@ public function getMessage()
             ->setBcc($this->getBcc())
             ->setReplyTo($this->getReplyTo())
             ->setBody($this->getBody(), 'text/html');
-
+        $messageHeaders = $message->getHeaders();
+        $dbHeaders = $this->getHeaders();
+        if (count($dbHeaders)) {
```
Comment (lenybernard) [inline Entity/Mail.php]: no need to count I think right ?

File: Entity/Mail.php [hunk 3]
```diff
@@ -408,6 +425,37 @@ public function setStatus($status)
...
+     * @param HeaderSet $headers The headers of the email
+     *
+     * @return Mail
+     */
+    public function setHeaders($headers)
+    {
+        $headerNames = $headers->listAll();
+        $headersArray = [];
+        foreach($headerNames as $headerName) {
+          if (!$this->isDefaultHeader($headerName) && $headers->get($headerName) instanceof \Swift_Mime_Headers_UnstructuredHeader) {
```
Comment (lenybernard) [inline Entity/Mail.php]: Can you tell me why you're checking both ? I guess a simple `Swift_Mime_Headers_UnstructuredHeader` type check would be sufficient, isn't it ?

File: Entity/Mail.php (hunk 2 above)
Comment (TristanPouliquen) [reply Entity/Mail.php]: It was failing when there were no headers in db. Changed failback to an array casting in getter method.

**Pull request merged**